tools/test/test_codegen_model.py

```diff
@@ -158,6 +159,12 @@ def test_parse_custom_class_type(self) -> None:
         self.assertEqual(custom_class_name, custom_class_type.class_name)
         self.assertEqual(custom_class_name_with_prefix, str(custom_class_type))
 
+    def test_rename_dispatch_key_privateuse1(self) -> None:
+        custom_device_name = "FOO"
+        rename_dispatch_key_privateuse1(custom_device_name)
+        self.assertEqual(str(DispatchKey.PrivateUse1), custom_device_name)
+        self.assertEqual(DispatchKey.parse(custom_device_name), DispatchKey.PrivateUse1)
```

Comment:
You can't do the unit test like this, you're monkey patching a bunch of attributes and then failing to reset it when you're done with the test.

Reply:
Yeap, this's my fault. I have saved the original pointer and later restored it. @ezyang  Could you take another look?